compiler/src/iree/compiler/Codegen/Common/FlattenMemRefs.cpp

```diff
@@ -164,185 +164,151 @@ static bool checkLayout(Value val) {
 }
 
 namespace {
-struct FlattenMemrefLoad : public OpRewritePattern<memref::LoadOp> {
-  using OpRewritePattern::OpRewritePattern;
+template <typename T>
+static Value getTargetMemref(T op) {
```

Comment:
Nice! We can probably do the same for the `replaceOp`

Reply:
That is a good idea actually.